optuna/trial/_trial.py

```diff
@@ -136,6 +136,12 @@ def objective(trial):
             high:
                 Upper endpoint of the range of suggested values. ``high`` is excluded from the
                 range.
+
+                .. note::
+                    If ``step`` is specified, ``high`` is included as well as ``low`` because
+                    this method falls back to :func:`~optuna.trial.Trial.suggest_discrete_uniform`
+                    with ``step`` argument.
```

Comment:
A nit but the following might be simpler and more concise (not having to be confused whether you didn't mean `q` or not). Sorry but I did not check the line length.
```suggestion
                    this method falls back to :func:`~optuna.trial.Trial.suggest_discrete_uniform`.
```